io-engine/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -1153,4 +1159,74 @@ impl<'n> Nexus<'n> {
             }
         });
     }
+
+    /// Resets nexus nvme children in case of initiator timeout.
+    pub(crate) async fn reset_all_children(nexus_name: String) {
+        let mut nex = match nexus_lookup_mut(&nexus_name) {
+            Some(v) => v,
+            None => {
+                warn!("Reset nexus children request: '{nexus_name}' not found");
+                return;
+            }
+        };
+
+        info!("{nex:?}: resetting nexus children ...");
+
+        // Pause the nexus.
+        if let Err(e) = nex.as_mut().pause().await {
+            error!("{nex:?}: failed to pause: {e}");
+            return;
+        }
+
+        debug!("{nex:?}: paused for children reset");
+
+        for child in nex.children_iter() {
+            if !(child.is_healthy() || child.is_opened_unsync()) {
+                continue;
+            }
+
+            let dev = match child.get_device() {
+                Ok(dev) => dev,
+                Err(e) => {
+                    error!("{child:?}: failed to get device: {e:?}");
+                    continue;
+                }
+            };
+
+            if dev.driver_name() != "nvme" {
+                debug!("{child:?}: driver is not NVMe-oF, skipping reset");
```

Comment:
Why do we skip for local children?

Reply:
Ah I see, because we use nvmf code directly?

Reply:
I guess, yes. I'm not really familiar how this is supposed to work. We disabled Nexus reset is disabled by default, as it never actually worked properly.